wordpress-mobile/WordPress-iOS
# Environment

This PR adds a new global called `Environment`. 

The `Environment` is meant to gather all globals / singletons used in the app that define the global state of the "world". In this way, is easier to take control of our app Environment when we need to Test.

Ideally we can start implementing and using this `Environment` in small steps:

- If we need to access a global value or send a message to a global entity, we should try to reach it through the `Environment.current`. 
- If it doesn't exist we can add it to the Environment:
  - First we declare a protocol that define the global's interface (its enough to declare just what you need, or the most common use cases)
  - Then we make the global conform to that interface. 
  - Finally we can add a property to the Environment following the example of the properties already there.

**For testing:**

We can create doubles that implement the needed protocol, and then we call `Environment.replaceEnvironment(valueToReplace: MyDouble())`.

**To test this PR:**
- Build and run.
- Make sure that the changes on `NotificationDetailsViewController.swift` where the `Environment` is used are working properly (it calls the object it should).
